test/irmin-pack/test_snapshot.ml

```diff
@@ -139,6 +139,7 @@ let test_in_memory () =
   let* tree2 = tree2 () in
   let* () = test tree2 3 in
   let* () = S.Repo.close repo_export in
+  S.flush repo_import;
   S.Repo.close repo_import
```

Comment:
The idea it should never be necessary for an irmin-pack user to call flush because we always take care of it for them. `batch` flushes before returning. I think that `Snapshot.Import.close` ( https://github.com/mirage/irmin/blob/main/src/irmin-pack/unix/snapshot.ml#L329) should flush too.

I can make the diff if you want